### DISCUSSION TITLE
collections.Hashable fails in distribute environment

### DISCUSSION THREAD

**Discussion by awolff64** [coreemu/core#756] [Q&A] [answered]
Hi,

I am trying a distributed setup.
When I configure a node on a remote I get an error:
collections has no 'Hashable'

I traced down the problem to the file:

./lib/python3.10/site-packages/invoke/vendor/yaml3/constructor.py

and changed 
            if not isinstance(key, collections.Hashable):
to
            if not isinstance(key, collections.**abc**.Hashable):

=> works now.

Had seen such a solution in other contexts.
However I do not know/understand where a proper solution for this resides.
Is it coreemu? Is is a dependency? Is is a base library?

**Comment by bharnden** [comment]
When you say, I am trying a distributed setup. What exactly does that mean? What OS? What exact steps are you taking?

Looking around this appears to likely be related to Python 3.10+ formally removing the deprecated import of `collections.Hashable`

**Reply by awolff64** [reply]
I have setup up a server running a daeon (lxc container, ubuntu 22.04, host ubuntu 22.10)

I run the "main" with a deamon and the gui on ubuntu 22.04 VM. This works fine as long as I do not use a server.
In the gui I defined a server named "WAN" and defined a node (router) on the WAN server.
When I press the StartSession button I got the reported error which I fixed with the ".abc."

As you can see I run python 3.10.
Looks like the yaml3 code is not under your responsibility but form the py-environment.?

**Comment by bharnden** [comment, marked as answer]
Appears to be an issue specifically with older versions of invoke due to python 3.10 removing the deprecated code.

You can use your hotfix, or use python 3.9 for now. I will look to update invoke to help alleviate this issue for the future.

**Reply by awolff64** [reply]
Thanks for the explanations - I will keep on working with my hotfix.
